girder_worker_utils/transforms/girder_io.py

```diff
@@ -117,3 +119,34 @@ def cleanup(self):
                 shutil.rmtree(self.output_file_path)
             else:
                 os.remove(self.output_file_path)
+
+
+class GirderUploadJobArtifact(GirderClientResultTransform):
+    def __init__(self, job_id=None, name=None, **kwargs):
+        super(GirderUploadJobArtifact, self).__init__(**kwargs)
+        self.job_id = job_id
+        self.name = name
+
+    def _repr_model_(self):
+        return "{}('{}')".format(self.__class__.__name__, self.job_id)
+
+    def _upload_artifact(self, file):
+        qs = urlencode({
+            'name': self.name or os.path.basename(file),
+            'size': os.stat(file).st_size,
+            'mimeType': mimetypes.guess_type(file)[0]
+        })
+        with open(file, 'rb') as fh:
+            self.gc.post('job/%s/artifact?%s' % (self.job_id, qs), data=fh)
+
+    def transform(self, path):
+        if self.job_id is None:
+            self.job_id = str(self.job['_id'])
+
+        if os.path.isdir(path):
+            for f in os.listdir(path):
+                f = os.path.join(path, f)
+                if os.path.isfile(f):
```

Comment:
Won't this flat-out ignore nested directories? That's fine if that is the intended functionality,  but either a comment or something in the class docstring would be great making that explicit.

Reply:
It is intentional, for now. I can add a comment to make that clear.